crates/uv-toolchain/src/version_files.rs

```diff
@@ -0,0 +1,67 @@
+use fs_err as fs;
+use std::{io, path::PathBuf};
+use tracing::debug;
+
+use crate::ToolchainRequest;
+
+/// Read [`ToolchainRequest`]s from a version file, if present.
+///
+/// Prefers `.python-versions` then `.python-version`.
+/// If only one Python version is desired, use [`request_from_version_files`] which prefers the `.python-version` file.
+pub async fn requests_from_version_files() -> Result<Option<Vec<ToolchainRequest>>, io::Error> {
```

Comment:
Should those be methods on `ToolchainRequest`?

Reply:
That's a good question, I guess they could be? It seems a little weird to me to return `Vec<Self>`?

Reply:
I could go either way personally.

Does this need to be `Option<Vec<T>>` though? Can we just use `Vec<T>`? I think the former is necessary if you specifically need to distinguish between "not specified" and "empty sequence."

Reply:
We want to distinguish between the file does not exist and the file exists but is empty.

Reply:
I'll save moving these methods for after the stack is merged (if it makes sense) to save myself some rebasing.